From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MinRi2 <2275045670@qq.com>
Date: Sat, 20 Apr 2024 20:34:21 +0800
Subject: [PATCH] =?UTF-8?q?C:=20=E6=96=B0=E5=88=B6=E6=A0=B8=E5=BF=83?=
 =?UTF-8?q?=E8=B5=84=E6=BA=90=E6=98=BE=E7=A4=BA?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

物品平均变化量;建造耗量重排

优化代码与显示；修复Plan灰框bug；移除物品平均值
way-zer <himc.wicp@gmail.com> on 2024/4/30 at 0:09

拆除建筑，续建建筑显示oh no图标
MinRi2 <2275045670@qq.com> on 2024/5/4 at 15:25

优化显示效果;更紧凑的Item布局
way-zer <himc.wicp@gmail.com> on 2024/5/17 at 20:50
---
 core/src/mindustry/ui/fragments/HudFragment.java | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/core/src/mindustry/ui/fragments/HudFragment.java b/core/src/mindustry/ui/fragments/HudFragment.java
index 6f8bb852edfe397c4abbbe9268f8b5648c19add3..37a57b425a423ddff029268c938423f6711340c8 100644
--- a/core/src/mindustry/ui/fragments/HudFragment.java
+++ b/core/src/mindustry/ui/fragments/HudFragment.java
@@ -30,6 +30,7 @@ import mindustry.ui.*;
 import mindustry.world.blocks.storage.*;
 import mindustry.world.blocks.storage.CoreBlock.*;
 import mindustryX.features.*;
+import mindustryX.features.ui.*;
 
 import static mindustry.Vars.*;
 import static mindustry.gen.Tex.*;
@@ -41,7 +42,7 @@ public class HudFragment{
     public boolean shown = true;
 
     private ImageButton flip;
-    private CoreItemsDisplay coreItems = new CoreItemsDisplay();
+    public NewCoreItemsDisplay coreItems = new NewCoreItemsDisplay();
 
     private String hudText = "";
     private boolean showHudText;
@@ -89,11 +90,6 @@ public class HudFragment{
             showToast(Icon.warning, Core.bundle.format("sector.attacked", e.sector.name()));
         });
 
-        Events.on(ResetEvent.class, e -> {
-            coreItems.resetUsed();
-            coreItems.clear();
-        });
-
         //paused table
         parent.fill(t -> {
             t.name = "paused";
@@ -318,8 +314,9 @@ public class HudFragment{
             t.collapser(v -> v.add().height(pauseHeight), () -> state.isPaused() && !netServer.isWaitingForPlayers()).row();
 
             t.table(c -> {
+                c.touchable = Touchable.disabled;
                 //core items
-                c.top().collapser(coreItems, () -> Core.settings.getBool("coreitems") && !mobile && shown).fillX().row();
+                c.top().add(coreItems).fillX().row();
 
                 float notifDuration = 240f;
                 float[] coreAttackTime = {0};
